### PR TITLE
Update symbol grammar

### PR DESCRIPTION
**Pull request by moxaj**
### Description of the Change

See the [spec](https://clojure.org/reference/reader):

> Symbols begin with a non-numeric character <...>
> Symbols beginning or ending with ':' are reserved by Clojure <...>

This PR changes the grammar associated with symbols, disallowing leading digits and colons, and also allowing single-character symbols.

### Alternate Designs

No alternatives were considered, the change seems straightforward.

### Benefits

This fixes a highlighting issue when a keyword/number following a `def*` symbol would be highlighted as a symbol (`entity.global.clojure`) (see #62).

### Possible Drawbacks

None that I can think of.

### Applicable Issues

#62

### CODE REVIEW

**Comment by moxaj** [comment]
Edited the description to reflect the changes in the second commit.

**Comment by 50Wliu** [comment]
Would you mind adding a spec for this please?

**Comment by moxaj** [comment]
@50Wliu any ETA on this?

**Comment by bjonica** [comment]
I would love to see this merged. Not recognizing single-character symbols makes syntax coloring look really messy.

**Comment by lee-dohm** [comment]
@50Wliu if you're cool with this change, we are.

File: spec/clojure-spec.coffee [hunk 1]
```diff
@@ -160,6 +163,12 @@ describe "Clojure grammar", ->
     expect(tokens[1]).toEqual value: "/", scopes: ["source.clojure"]
     expect(tokens[2]).toEqual value: "bar", scopes: ["source.clojure", "meta.symbol.clojure"]
 
+    {tokens} = grammar.tokenizeLine "x"
+    expect(tokens[0]).toEqual value: "x", scopes: ["source.clojure", "meta.symbol.clojure"]
+
+    {tokens} = grammar.tokenizeLine "1foobar"
+    expect(tokens[0]).not.toEqual value: "1foobar", scopes: ["source.clojure", "meta.symbol.clojure"]
```
Comment (50Wliu) [inline spec/clojure-spec.coffee]: Can this be changed to an explicit `.toEqual`?  That way if the `meta.symbol.clojure` scope changes this test will continue to fail if the behavior regresses.

**Pull request merged**